### PR TITLE
[core] Update babel configs

### PR DESCRIPTION
**Pull request by romgrk**
Clean up the babel configs, and share as much as possible between the root & docs configs.

Linked: https://github.com/mui/material-ui/pull/37461

### CODE REVIEW

File: docs/babel.config.js [hunk 1]
```diff
@@ -1,65 +1,44 @@
...
-    /@babel[\\|/]runtime/,
-    // Fix const foo = /{{(.+?)}}/gs; crashing.
-    /prettier/,
-    /@mui[\\|/]docs[\\|/]markdown/,
-  ],
-  env: {
-    production: {
-      plugins: [
-        // TODO fix useGridSelector side effect and enable back.
-        // '@babel/plugin-transform-react-constant-elements',
```
Comment (romgrk) [inline docs/babel.config.js:59]: @oliviertassinari I've uncommented this one to see if we can cleanup, not sure what's the side-effect though.

File: docs/babel.config.js [hunk 2]
```diff
@@ -1,65 +1,44 @@
...
+        'babel-plugin-module-resolver',
+        {
+          alias,
+          transformFunctions: ['require', 'require.context'],
+        },
+      ],
     ],
-    'babel-plugin-optimize-clsx',
-    // for IE 11 support
-    '@babel/plugin-transform-object-assign',
```
Comment (romgrk) [inline docs/babel.config.js:38]: IIRC, we don't support IE11 anymore.

File: docs/babel.config.js [hunk 3]
```diff
@@ -1,65 +1,44 @@
...
+    ...getBaseConfig.getDefaultAlias(),
+    '@mui/monorepo': '../node_modules/@mui/monorepo',
+    '@mui/docs': '../node_modules/@mui/monorepo/packages/mui-docs/src',
+    docsx: './',
+  };
 
-module.exports = {
-  presets: [
-    // backport of https://github.com/zeit/next.js/pull/9511
-    ['next/babel', { 'transform-runtime': { corejs: 2, version: transformRuntimeVersion } }],
```
Comment (romgrk) [inline docs/babel.config.js:27]: Latest corejs version is `3` and this is 4 years old, I think we can use the default version?

@eps1lon

File: docs/babel.config.js (hunk 1 above)
Comment (oliviertassinari) [reply docs/babel.config.js:59]: I have updated the PR (#666) to better describe the issue.

File: docs/babel.config.js (hunk 2 above)
Comment (oliviertassinari) [reply docs/babel.config.js:38]: Depends, it's also about Safari.

Comment (LukasTy) [reply docs/babel.config.js:38]: Is it? Do we support Safari <v9?
It does seem to be supported [starting v9](https://caniuse.com/mdn-javascript_builtins_object_assign) on Safari and iOS Safari. 🤔

**Pull request closed without merging**